src/ol/style/regularshape.js

```diff
@@ -401,11 +431,12 @@ ol.style.RegularShape.prototype.draw_ = function(renderOptions, context, x, y) {
   context.translate(x, y);
 
   context.beginPath();
+  var points = this.points_;
   if (this.radius2_ !== this.radius_) {
-    this.points_ = 2 * this.points_;
+    points = 2 * points;
```

Comment:
If I understand that correct you could have done the following alternatively (in `clone`), right:

```js
var style = new ol.style.RegularShape({
  // …
  points: this.radius2_ !== this.radius_ ? this.getPoints() / 2 : this.getPoints()
  // …
});
```

This way we wouldn't need the above changes, right?